### DISCUSSION TITLE
SyntaxError when running `npm start` following Jokes App tutorial

### DISCUSSION THREAD

**Discussion by drazik** [remix-run/remix#2611] [Q&A]
Hello,

I am trying to follow the Jokes App tutorial. I created the app using `npx create-remix@latest`, then built it using `npm run build`. For these two steps everything is OK. But when I execute `npm start`, I get the following error:

```console
> start
> node server.js

(node:98502) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/home/drazik/workspace/perso/ecoledesclaviers/edc/server.js:1
import { createRequestHandler } from "@remix-run/vercel";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1031:15)
    at Module._compile (node:internal/modules/cjs/loader:1065:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
```

Following the hint in the error message, I tried to add `"type": "module"` in the `package.json` file. But this time I get:

```console
> start
> node server.js

node:internal/process/esm_loader:94
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/drazik/workspace/perso/ecoledesclaviers/edc/node_modules/@remix-run/dev/server-build' imported from /home/drazik/workspace/perso/ecoledesclaviers/edc/server.js
Did you mean to import @remix-run/dev/server-build.js?
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:416:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

The `@remix-run/dev/server-build.js` module exists so I don't know what's wrong.

I am using Node v16.13.1, npm v8.1.2. When I created the app using remix CLI I choosed to deploy the app on Vercel and to use TypeScript.

Let me know if I did something wrong, if I forgot some valuable details about my setup or whatever that can help in finding what's wrong.

Thanks a lot, have a nice day :v:

**Comment by komrihe** [comment]
can some help me out with the above errors

**Reply by kiliman** [reply]
It looks like the OP chose the Vercel Adapter. I think there as a bug in that template back in April that has since been fixed.

Which adapter did you choose? The tutorial suggests you use Remix App Server.

**Reply by runoncedev** [reply]
> It looks like the OP chose the Vercel Adapter. I think there as a bug in that template back in April that has since been fixed.
> 
> Which adapter did you choose? The tutorial suggests you use Remix App Server.

I'm using Vercel and having exactly the same problem.

Any ideas?

This are my dependencies:

```
blog@ /home/runo/Devv/blog
├── @prisma/client@4.2.1
├── @remix-run/dev@1.6.8
├── @remix-run/eslint-config@1.6.8
├── @remix-run/node@1.6.8
├── @remix-run/react@1.6.8
├── @remix-run/serve@1.6.8
├── @remix-run/vercel@1.6.8
├── @types/react-dom@17.0.17
├── @types/react@17.0.48
├── @vercel/node@2.5.8
├── eslint@8.22.0
├── prisma@4.2.1
├── react-dom@17.0.2
├── react@17.0.2
└── typescript@4.7.4
```

**Reply by nikitapashinsky** [reply]
I have the same exact issue… Has anyone been able to find out how to fix this? Thank you!

**Reply by machour** [reply]
Can you publish a repository reproducing the issue ?

**Reply by nikitapashinsky** [reply]
Hi @machour, here's a repository with my settings. I tried to outline the issue in the README. 
_[EDIT: removed link because the repo / configuration is not the issue]_

Thank you for your help!

**Reply by machour** [reply]
Could it be that you want to run `npm run dev` locally, instead of `npm run build` & `npm run start` that are meant for production (vercel) ?

**Reply by nikitapashinsky** [reply]
That's probably it... 🤦‍♂️ Sorry, I'm very new to all of this. I don't need to run `npm run start` locally because I can simply use the dev server. And the app actually deploys to Vercel just fine. Thank for your help again, sorry about the stupid question haha.
